### PR TITLE
Fix generate password functionality

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose
- Fix bugs with generate password functionality on add user wizard.
- Fix ESLint warnings

### Related Issues
- https://github.com/wso2/product-is/issues/12320

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/users/components/add-user.tsx [hunk 1]
```diff
@@ -78,7 +78,7 @@ export const AddUser: React.FunctionComponent<AddUserProps> = (props: AddUserPro
     ] = useState<string>(PRIMARY_USERSTORE_PROPERTY_VALUES.UsernameJavaScriptRegEx);
     const [ isUsernameRegExLoading, setUsernameRegExLoading ] = useState<boolean>(false);
     const [ password, setPassword ] = useState<string>("");
-    const [ passwordScore, setPasswordScore ] = useState<number>(-1);
+    const [ _passwordScore, setPasswordScore ] = useState<number>(-1);
```
Comment (thivi) [inline apps/console/src/features/users/components/add-user.tsx:81]: If we are not using this anywhere, we can remove this, right?

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/users/utils/generate-password.utils.ts:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Utility functions for password generation on user creation
+ */
+
```
Comment (thivi) [inline apps/console/src/features/users/utils/generate-password.utils.ts:22]:
```suggestion
```

**Review by thivi — APPROVED** [review]

File: apps/console/src/features/users/components/add-user.tsx [hunk 4]
```diff
@@ -169,35 +169,35 @@ export const AddUser: React.FunctionComponent<AddUserProps> = (props: AddUserPro
...
     const handlePasswordChange = (values: Map<string, FormValue>): void => {
         const password: string = values.get("newPassword").toString();
+
         setPassword(password);
     };
 
-    /**
-     * The following function generate a random password.
-     */
     const generateRandomPassword = (): void => {
```
Comment (brionmario) [inline apps/console/src/features/users/components/add-user.tsx:176]: Shall we add the doc comment back.

File: apps/console/src/features/users/components/add-user.tsx [hunk 5]
```diff
@@ -564,7 +563,7 @@ export const AddUser: React.FunctionComponent<AddUserProps> = (props: AddUserPro
                     </Grid.Column>
                 </Grid.Row>
                 { emailVerificationEnabled &&
```
Comment (brionmario) [inline apps/console/src/features/users/components/add-user.tsx:565]: Reformat

```
{ emailVerificationEnabled && (
    <Grid.Row columns={ 1 }>
    </Grid.Row>
) }

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const generateRandomNumbers = (upperLimit: number): number => {
```
Comment (brionmario) [inline apps/console/src/features/users/utils/generate-password.utils.ts:19]: Lets add doc comments to the functions.

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    return Math.random() * (upperLimit);
+};
+
+const removeIgnoredCharacters = (str: string, ignoredChars: string[]) => {
```
Comment (brionmario) [inline apps/console/src/features/users/utils/generate-password.utils.ts:23]:
```suggestion
const removeIgnoredCharacters = (str: string, ignoredChars: string[]): string => {
```

File: apps/console/src/features/users/components/add-user.tsx (hunk 1 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/components/add-user.tsx]: Fixed with 0b225bb8b2f850c71ab6ffb8b5b75aed28d54650

File: apps/console/src/features/users/utils/generate-password.utils.ts (hunk 3 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts]: Fixed with 0b225bb8b2f850c71ab6ffb8b5b75aed28d54650

File: apps/console/src/features/users/utils/generate-password.utils.ts (hunk 2 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts]: Fixed with 0b225bb8b2f850c71ab6ffb8b5b75aed28d54650

File: apps/console/src/features/users/components/add-user.tsx (hunk 4 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/components/add-user.tsx:179]: Fixed with f1efba513d6385145463c2f9f6462bf4a21f6b17

File: apps/console/src/features/users/components/add-user.tsx (hunk 5 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/components/add-user.tsx]: Fixed with f1efba513d6385145463c2f9f6462bf4a21f6b17

File: apps/console/src/features/users/utils/generate-password.utils.ts (hunk 6 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts:24]: Fixed with f1efba513d6385145463c2f9f6462bf4a21f6b17

File: apps/console/src/features/users/utils/generate-password.utils.ts (hunk 7 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts]: Fixed with f1efba513d6385145463c2f9f6462bf4a21f6b17

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+ * @param {boolean} isAlphabetsLowercaseAllowed - whether to include lowercase alphabets
+ * @param {boolean} isAlphabetsUppercaseAllowed - whether to include uppercase alphabets
+ * @param {boolean} isNumbersAllowed - whether to include numbers
+ * @param {boolean} isSpecialCharactersAllowed - whether to include special characters
+ * @param {number} minAlphabetsLowercase - required minimum number of lowercase alphabets
+ * @param {number} minAlphabetsUppercase - required minimum number of uppercase alphabets
+ * @param {number} minNumbers - required minimum number of numbers
+ * @param {number} minSpecialCharacters - required minimum number of special charaters
+ * @param {string[]} ignoredCharactors - array of charaters to be excluded
+ * @returns {string} - generated password
```
Comment (Yasara-Y) [inline apps/console/src/features/users/utils/generate-password.utils.ts:54]: Add a full-stop at the end of each comment.

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+ * @param {number} upperLimit - upperlimit (exclusive) for random numbers range
+ * @returns {number}
+ */
+const generateRandomNumbers = (upperLimit: number): number => {
+    return Math.random() * (upperLimit);
+};
+
+/**
+ * The following function is to remove an array of strings from a string
+ * @param {string} str - original string
```
Comment (Yasara-Y) [inline apps/console/src/features/users/utils/generate-password.utils.ts:30]: Leave a line after the method description. Check other places too.

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+ * The following function is for generating random numbers in range of [0 - upperlimit)
+ * @param {number} upperLimit - upperlimit (exclusive) for random numbers range
+ * @returns {number}
+ */
+const generateRandomNumbers = (upperLimit: number): number => {
+    return Math.random() * (upperLimit);
+};
+
+/**
+ * The following function is to remove an array of strings from a string
```
Comment (Yasara-Y) [inline apps/console/src/features/users/utils/generate-password.utils.ts:29]: Leave a line after the method description. Check other places too.

File: apps/console/src/features/users/utils/generate-password.utils.ts (hunk 8 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts:54]: Fixed with b017ea7b1473e1a5ff47f46cec10a6963c118668

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+ * @param {string} str - original string
+ * @param {string[]} ignoredChars - array of strings to be removed
+ * @returns {string} - modified original string by removing ignoredChars
+ */
```
[the remark this answers is not in the document]
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts:33]: Fixed with b017ea7b1473e1a5ff47f46cec10a6963c118668

File: apps/console/src/features/users/utils/generate-password.utils.ts (hunk 10 above)
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts:29]: Fixed with b017ea7b1473e1a5ff47f46cec10a6963c118668

File: apps/console/src/features/users/utils/generate-password.utils.ts [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * The following function is for generating random numbers in range of [0 - upperlimit)
```
[the remark this answers is not in the document]
Comment (JayaShakthi97) [reply apps/console/src/features/users/utils/generate-password.utils.ts:20]: Fixed with b017ea7b1473e1a5ff47f46cec10a6963c118668

**Pull request merged**